pennylane_lightning/src/algorithms/AdjointDiff.hpp

```diff
@@ -291,44 +305,51 @@ template <class T> class OpsData {
      *
      * @return size_t Number of operations.
      */
-    size_t getSize() const { return ops_name_.size(); }
+    [[nodiscard]] auto getSize() const -> size_t { return ops_name_.size(); }
```

Comment:
Not sure how `[[nodiscard]]` would be helpful except making some dev challenges to ensure not-discarding the return value. I think, the `[[nodiscard]]` attribute is more practical where discarding the result leads to a computational error.